### PR TITLE
[gwctl]Optimize Print in backends

### PR DESCRIPTION
**Pull request by yeedove**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Optimize print and reduce repeated logic. Maintain agreement with print methods of other resources
```

### CODE REVIEW

File: gwctl/pkg/printer/backends.go [hunk 1]
```diff
@@ -49,59 +49,36 @@ func (bp *BackendsPrinter) Print(resourceModel *resourcediscovery.ResourceModel)
 		os.Exit(1)
 	}
 
-	backendNodes := make([]*resourcediscovery.BackendNode, 0, len(resourceModel.Backends))
-	for _, backendNode := range resourceModel.Backends {
-		backendNodes = append(backendNodes, backendNode)
-	}
+	backends := common.MapToValues(resourceModel.Backends)
```
Comment (gauravkghildiyal) [inline gwctl/pkg/printer/backends.go:52]: Perhaps we can start using the same `maps.Values()` here as well (and eventually get rid of common.MapToValues in the other places as well -- you need not make all the other changes here)

**Review by gauravkghildiyal — COMMENTED** [review]
Thanks @yeedove.

Depending on when https://github.com/kubernetes-sigs/gateway-api/pull/3051 merges, you may (or may not have to rebase this)

File: gwctl/pkg/printer/backends.go (hunk 1 above)
Comment (yeedove) [reply gwctl/pkg/printer/backends.go]: done!

File: gwctl/pkg/printer/backends_test.go [hunk 2]
```diff
@@ -198,9 +198,9 @@ func TestBackendsPrinter_Print(t *testing.T) {
 
 	got := params.Out.(*bytes.Buffer).String()
 	want := `
-NAMESPACE  NAME       TYPE     REFERRED BY ROUTES                                AGE  POLICIES
-ns1        foo-svc-1  Service  ns1/foo-httproute-1                               3d   1
-ns1        foo-svc-2  Service  ns1/foo-httproute-2,ns1/foo-httproute-3 + 2 more  2d   0
+NAMESPACE  NAME       TYPE     REFERRED BY ROUTES                               AGE  POLICIES
+ns1        foo-svc-1  Service  ns1/foo-httproute-1                              3d   1
+ns1        foo-svc-2  Service  ns1/foo-httproute-2,ns1/foo-httproute-3,+2 more  2d   0
```
Comment (gauravkghildiyal) [inline gwctl/pkg/printer/backends_test.go:203]: Actually, not having a comma after `ns1/foo-httproute-3` is better.

**Comment by gauravkghildiyal** [comment]
Thanks @yeedove!

/lgtm
/approve

**Comment by gauravkghildiyal** [comment]
This requires some lint fixes to pass

**Pull request merged**